core/src/test/java/org/infinispan/security/DynamicRBACTest.java

```diff
@@ -107,6 +114,33 @@ public void testClusterPermissionMapper() {
       assertEquals(1, roles.size());
    }
 
+   public void testJoiner() throws PrivilegedActionException {
+      crm.grant("wizard", "gandalf");
+      Cache<?, ?> crmCache = extractField(crm, "clusterRoleMap");
+      ClusterTopologyManager crmTM = TestingUtil.extractComponent(crmCache, ClusterTopologyManager.class);
+      crmTM.setRebalancingEnabled(false);
+      await(cpm.addRole(Role.newRole("rogue", true, AuthorizationPermission.LISTEN)));
+      Cache<?, ?> cpmCache = extractField(cpm, "clusterPermissionMap");
+      ClusterTopologyManager cpmTM = TestingUtil.extractComponent(cpmCache, ClusterTopologyManager.class);
+      cpmTM.setRebalancingEnabled(false);
+      EmbeddedCacheManager joiner = Security.doAs(ADMIN, (PrivilegedExceptionAction<EmbeddedCacheManager>) () ->
+            addClusterEnabledCacheManager(getGlobalConfigurationBuilder(), getConfigurationBuilder(), new TransportFlags().withFD(true))
+      );
+      ClusterRoleMapper joinerCrm = Security.doAs(ADMIN, (PrivilegedExceptionAction<ClusterRoleMapper>) () -> (ClusterRoleMapper) joiner.getCacheManagerConfiguration().security().authorization().principalRoleMapper());
+      Set<String> admin0 = crm.list("gandalf");
+      assertTrue(admin0.contains("wizard"));
+      Set<String> admin2 = joinerCrm.list("gandalf");
+      assertFalse(admin2.contains("wizard"));
```

Comment:
worth testing that `joinerCrm.principalToRoles()` does not have `wizard` in the set (i.e., method `principalToRoles` should not do a remote request)

Reply:
Done